File: BleModule/src/main/java/com/topdon/ble/util/DefaultLogger.java
package com.topdon.ble.util;


public class DefaultLogger implements Logger {
    private final String tag;
    private boolean isEnabled;

    public DefaultLogger(String tag) {
        this.tag = tag;
    }

    @Override
    public boolean isEnabled() {
        return isEnabled;
    }

    @Override
    public void setEnabled(boolean isEnabled) {
        this.isEnabled = isEnabled;
    }

    @Override
    public void log(int priority, int type, String msg) {
    }

    @Override
    public void log(int priority, int type, String msg, Throwable th) {
    }
}

File: ble-core/src/main/java/com/mpdc4gsr/ble/core/util/DefaultLogger.kt
package com.mpdc4gsr.ble.core.util

import android.util.Log

class DefaultLogger(private val tag: String?) : Logger {
    override var isEnabled = false

    override fun log(priority: Int, type: Int, msg: String?) {
        if (isEnabled) {
            Log.println(priority, tag, msg!!)
        }
    }

    override fun log(priority: Int, type: Int, msg: String?, th: Throwable?) {
        if (isEnabled) {
            if (msg != null) {
                log(priority, type, msg + "\n" + Log.getStackTraceString(th))
            } else {
                log(priority, type, Log.getStackTraceString(th))
            }
        }
    }
}
